ccdproc/image_collection.py

```diff
@@ -62,14 +62,35 @@ class ImageFileCollection(object):
         The filenames are assumed to be in ``location``.
         Default is ``None``.
 
+    glob_include: str, optional
+        Unix-style filename pattern to select filenames to include in the file
+        collection. Can be used in conjunction with ``glob_exclude`` to
+        easily select subsets of files in the target directory.
+
+    glob_exclude: str, optional
+        Unix-style filename pattern to select filenames to exclude from the
+        file collection. Can be used in conjunction with ``glob_include`` to
+        easily select subsets of files in the target directory.
+
     Raises
     ------
     ValueError
         Raised if keywords are set to a combination of '*' and any other
         value.
     """
     def __init__(self, location=None, keywords=None, info_file=None,
-                 filenames=None):
+                 filenames=None, glob_include=None, glob_exclude=None):
+
+        # Include or exclude files from the collection based on glob pattern
+        # matching - has to go above call to _get_files()
+        if glob_exclude is not None:
+            glob_exclude = str(glob_exclude) # some minimal validation
```

Comment:
`str` almost never fails, at least on Python3. Wouldn't it make more sense to let it fail when it actually does the `fnmatch`? Also the exception message when `str` fails is probably not helpful either so I don't see a compelling argument for this "minimal validation".

Or am I misunderstanding this?